lib/internal/buffer_new.js

```diff
@@ -0,0 +1,1014 @@
+'use strict';
+
+const binding = process.binding('buffer');
+const util = require('util');
+
+exports.Buffer = Buffer;
+exports.SlowBuffer = SlowBuffer;
+exports.INSPECT_MAX_BYTES = 50;
+
+
+Buffer.poolSize = 8 * 1024;
+var poolSize, poolOffset, allocPool;
+
+
+function createPool() {
+  poolSize = Buffer.poolSize;
+  allocPool = binding.create(poolSize);
+  poolOffset = 0;
+}
+
+
+function Buffer(arg) {
+  // Common case.
+  if (typeof arg === 'number') {
+    // If less than zero, or NaN.
+    if (arg < 0 || arg !== arg)
+      arg = 0;
+    return allocate(arg);
+  }
+
+  // Slightly less common case.
+  if (typeof arg === 'string') {
+    return fromString(arg, arguments[1]);
+  }
+
+  // Unusual.
+  return fromObject(arg);
+};
+
+Buffer.prototype.__proto__ = Uint8Array.prototype;
+Buffer.__proto__ = Uint8Array;
+
+
+if (!process.useOldBuffer) {
+  binding.setupBufferJS(Buffer.prototype);
+  // Buffer prototype must be past before creating our first pool.
+  createPool();
+}
```

Comment:
This was me just being paranoid. Didn't want to chance `setupBufferJS()` to run twice (was debugging a segfault issue). Will remove.